steam/ext/dota2/client.py

```diff
@@ -0,0 +1,127 @@
+
+__all__ = (
+    "Client",
+    "Bot",
+)
+
+
+class Client(Client_):
+    """Represents a client connection that connects to Steam. This class is used to interact with the Steam API, CMs
+    and the Dota 2 Game Coordinator.
+
+    :class:`Client` is a subclass of :class:`steam.Client`, so whatever you can do with :class:`steam.Client` you can
+    do with :class:`Client`.
+    """
+
+    _APP: Final = DOTA2
+    _state: GCState  # type: ignore  # PEP 705
+
+    async def fetch_top_source_tv_games(
+        self,
+        *,
+        search_key: str = MISSING,
+        league_id: int = MISSING,
+        hero_id: int = MISSING,
+        start_game: Literal[0, 10, 20, 30, 40, 50, 60, 70, 80, 90] = 0,
+        game_list_index: int = MISSING,
+        lobby_ids: list[int] = MISSING,
+    ) -> list[CMsgGCToClientFindTopSourceTVGamesResponse]:
+        """Fetch Top Source TV Games.
+
+        This functionality is similar to game list in the Watch Tab of Dota 2 game application.
+        It fetches summary data for the currently on-going Dota 2 matches from the following categories:
+
+        * tournament games
+        * highest average mmr games
+        * specific lobbies, like friends' games by their lobby_ids
+
+        Note
+        -------
+        Note that the following documentation for keyword arguments to query against is rather observant
+        than from official. So, please, if you know more or description is incorrect, contact us.
+
+        Parameters
+        ----------
+        search_key : :class: `str`, optional
+            Unknown purpose.
+        league_id : :class: `int`, optional
+            `league_id` for the professional tournament.
+        hero_id : :class: `int`, optional
+            `hero_id` to filter results by, just like in-client Watch Tab feature.
+        start_game : :class: `Literal[0, 10, 20, 30, 40, 50, 60, 70, 80, 90]`, optional, by default 0
+            This argument controls how many responses the game coordinator should return.
+            For example, `start_game=0` returns a list with 1 Response,
+            `start_game=90` returns a list of 10 Responses.
+            Game coordinator fills each response in consequent manner with games that satisfy keyword arguments
+            or with currently live highest average MMR games in case no other argument was given.
+        game_list_index : :class: `int`, optional
+            Only get responses matching `game_list_index`. Responses from Game Coordinator change
+            from time to time. The `game_list_index` indicates that those responses belong to the same chunk.
+        lobby_ids : :class: `list[int]`, optional
+            Lobby ids to query against.
+
+        Returns
+        -------
+        list[dota_gcmessages_client_watch.CMsgGCToClientFindTopSourceTVGamesResponse]
+            The list of responses from Dota 2 Coordinator.
+            Those are grouped into grouped into chunks of 10 games.
```

Comment:
```suggestion
        The responses from the GC. They are grouped into grouped into chunks of 10. games.
```
Why are they chunked into 10, and can we not just return as a model that isn't directly from the GC?

Reply:
I guess my wording sucks. It is just how this protobuf message works (by Valve). We send one message to GC and if for example
* `start_game=90` then game coordinator will send 10 messages back.
* if `start_game = 50` - it will send 6 messages back (Each message contains group-info about 10 games and some info about each game too. )
* if we requested `start_game=90, hero_id=1` and GC found 22 games then first 2 reponses will have 10 games, 3rd will have 2 and the rest is empty (but still have group-info).  

So I just add those responses in a list of responses to return

Idk, I kinda didn't want to go away from GC models or protobufs. I would rather want to have a clear way to use protobufs without rethinking Valve's structure or naming, like I basically send kwargs to the proto and get responses back.

Reply:
Am I correct in saying its start_game//10 + 1 sort of? I think if the messages are empty and don't contain any new data we shouldn't be returning them.

I really don't agree with a lot of Valve's namings of things which is a lot of the reason I created this library, very few methods (if any) return the protobufs directly because it just makes things harder to use especially because they don't have any sort of deprecation period or policy like I have now.